File: 240105/특정 문자로 끝나는 문자열/string-ending-with-specific-character.py
word = []
for i in range(10):
    word.append(input())

word2 = input()
cnt = 0 
for i in range(10):
    if word[i][-1]==word2:
        cnt+=1
        print(word[i])
if cnt == 0:
    print("None")

File: 240105/특정 문자로 끝나는 문자열/string-ending-with-specific-character.java
import java.util.Scanner;
public class Main {
    public static void main(String[] args) {
        // 여기에 코드를 작성해주세요.
        Scanner sc = new Scanner(System.in);
        String[] word = new String[10];

        for (int i = 0; i<10; i++){
            word[i] = sc.next();
        }
        char wor = sc.next().charAt(0);
        int cnt = 0 ;

        for(int i = 0 ; i<10; i++){
            int len = word[i].length();
            if (word[i].charAt(len - 1)==wor){
                System.out.println(word[i]);
                cnt++;
            }
        }
        if (cnt == 0 ){
            System.out.print("None");
        }

    }
}